icu4c/source/i18n/persncal.cpp

```diff
@@ -62,6 +65,45 @@ static const int32_t kPersianCalendarLimits[UCAL_FIELD_COUNT][4] = {
     {        0,        0,       11,       11}, // ORDINAL_MONTH
 };
 
+namespace { // anonymous
+
+static icu::UnicodeSet *gLeapCorrection = nullptr;
+static icu::UInitOnce gCorrectionInitOnce {};
+static int32_t gMinCorrection;
+}  // namespace
+U_CDECL_BEGIN
+static UBool calendar_persian_cleanup() {
+    if (gLeapCorrection) {
+        delete gLeapCorrection;
+        gLeapCorrection = nullptr;
+    }
+    gCorrectionInitOnce.reset();
+    return true;
+}
+U_CDECL_END
+
+namespace { // anonymous
+static void U_CALLCONV initLeapCorrection() {
+    static int16_t nonLeapYears[] = {
+       1502, 1601, 1634, 1667, 1700, 1733, 1766, 1799, 1832, 1865, 1898, 1931, 1964, 1997, 2030, 2059,
+       2063, 2096, 2129, 2158, 2162, 2191, 2195, 2224, 2228, 2257, 2261, 2290, 2294, 2323, 2327, 2356,
+       2360, 2389, 2393, 2422, 2426, 2455, 2459, 2488, 2492, 2521, 2525, 2554, 2558, 2587, 2591, 2620,
+       2624, 2653, 2657, 2686, 2690, 2719, 2723, 2748, 2752, 2756, 2781, 2785, 2789, 2818, 2822, 2847,
+       2851, 2855, 2880, 2884, 2888, 2913, 2917, 2921, 2946, 2950, 2954, 2979, 2983, 2987,
+    };
+    gMinCorrection = nonLeapYears[0];
+    icu::UnicodeSet prefab;
+    for (auto year : nonLeapYears) {
+        prefab.add(year);
+    }
+    gLeapCorrection = prefab.cloneAsThawed();
```

Comment:
```suggestion
    gLeapCorrection = prefab.cloneAsThawed().freeze();
```

should make contains() a lot faster (and makes the set immutable)

https://unicode-org.github.io/icu-docs/apidoc/released/icu4c/classicu_1_1UnicodeSet.html#ad8d781d46d63eb57eb24e372feef12ef